src/picongpu/include/simulationControl/MovingWindow.hpp

```diff
@@ -51,45 +51,90 @@ class MovingWindow
+                float_64(globalWindowSizeY) * (float_64(1.0) - slide_point)
+            );
+
+            /* Is the point when the virtual particle **passed** the GPU next to the last
+             * in the current step
+             */
+            const uint32_t firstSlideStep = math::floor(
+                float_64(subGrid.getGlobalDomain().size.y() - slidePointOffsetCells) * cellHeight / deltaWayPerStep
+            );
+
+            /* size of the moving window in pic length */
+            const float_64 windowRealSizeY = float_64(globalWindowSizeY - slidePointOffsetCells) * cellHeight;
+            /* Is the point when the virtual particle **passed** the moving window
+             * in the current step
+             */
+            const uint32_t firstMoveStep = math::floor(
+                windowRealSizeY / deltaWayPerStep
+            );
 
-            /* round to nearest cell to have smoother offset jumps */
-            if (offsetFirstGPU)
+            if (firstMoveStep <= currentStep)
             {
-                *offsetFirstGPU = math::floor(((float_64) stepsInLastGPU + stepsInFutureAfterComma) *
-                                              light_way_per_step / cell_height + 0.5);
+                /* calculate the current position of the virtual particle */
+                const float_64 currentVParticlePosition = deltaWayPerStep * float_64(currentStep);
+                const float_64 currentVParticleCell = currentVParticlePosition / cellHeight;
+                const uint32_t virtualParticlePosition = uint32_t(math::floor(currentVParticleCell)) + slidePointOffsetCells;
+
+                /* calculate the position of the virtual particle after the current step is calculated */
+                const float_64 nextVParticlePosition = deltaWayPerStep * float_64(currentStep + 1);
+                const float_64 nextVParticleCell = nextVParticlePosition / cellHeight;
+                /* This position is used to detect the point in time where the virtual particle
+                 * moves over a GPU border.
+                 */
+                const uint32_t virtualParticleNextPosition =  uint32_t(math::floor(nextVParticleCell)) + slidePointOffsetCells;
+
+                /* Moving window start moving if the virtual particle passed the border of
+                 * between two GPUs in the current step.
+                 * We slide if the virtual particle is in the next step on another GPU than in
+                 * the current step but not before the first slide step.
+                 */
+                if (firstSlideStep <= currentStep &&
+                   (virtualParticleNextPosition % gpuLocalSizeY) < (virtualParticlePosition % gpuLocalSizeY)
+                )
+                {
+                    incrementSlideCounter(currentStep);
+                    if (doSlide)
+                        *doSlide = true;
+                }
+
+                /* valid range for the offset is [0;GPU size in y direction) */
```

Comment:
add:
`since the moving window in PIConGPU always starts on the first plane (3D) / row (2D) of GPUs in y direction, this
is calculation is equal to the globalWindow.offset in y`